s3auth-relay/src/main/java/com/s3auth/relay/BaseFacade.java

```diff
@@ -0,0 +1,435 @@
+                }
+            }
+        } catch (final InterruptedException ex) {
+            Thread.currentThread().interrupt();
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    /**
+     * Shutdown a service.
+     * @param service The service to shut down
+     * @throws InterruptedException If fails to shutdown
+     */
+    private void shutdown(final ExecutorService service)
+        throws InterruptedException {
+        service.shutdown();
+        if (service.awaitTermination(Tv.TEN, TimeUnit.SECONDS)) {
+            Logger.info(this, "#shutdown(): succeeded");
+        } else {
+            Logger.warn(this, "#shutdown(): failed");
+            service.shutdownNow();
+            if (service.awaitTermination(Tv.TEN, TimeUnit.SECONDS)) {
+                Logger.info(this, "#shutdown(): shutdownNow() succeeded");
+            } else {
+                Logger.error(this, "#shutdown(): failed to stop threads");
+            }
+        }
+    }
+
+   /**
+    * Dispatcher for either {@link HttpThread} or {@link FtpThread}.
+    */
+    private static final class ThreadRunnableDispatcher implements Runnable {
+
+       /**
+        * The thread to run.
+        */
+        private final transient RequestDispatcher thread;
+
+       /**
+        * Package-private constructor.
+        * @param thrd The RequestDispatcher
+        */
+        ThreadRunnableDispatcher(final RequestDispatcher thrd) {
+            this.thread = thrd;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void run() {
+            try {
+                this.thread.dispatch();
+            } catch (final InterruptedException ex) {
+                Thread.currentThread().interrupt();
+                Logger.warn(this, "%s", ex);
+            }
+        }
+    }
+
+    /**
+     * HTTP facade (port listener).
+     *
+     * <p>The class is instantiated in {@link Main}, once per application run.
+     *
+     * <p>The class is immutable and thread-safe.
+     *
+     * @author Simon Njenga (simtuje@gmail.com)
+     * @author Yegor Bugayenko (yegor@tpc2.com)
+     * @version $Id$
+     * @since 0.1
+     * @see Main
+     */
+    @ToString
+    @SuppressWarnings("PMD.DoNotUseThreads")
+    @Loggable(Loggable.DEBUG)
+    protected static final class HttpFacade extends BaseFacade {
+
+       /**
+        * Base facade encapsulated.
+        */
+        private final transient BaseFacade facade;
+
+       /**
+        * Package-private constructor.
+        * @param hosts Hosts
+        * @param port Port number
+        * @param sslport SSL port number.
+        * @throws IOException If can't initialize
+        */
+        HttpFacade(@NotNull final Hosts hosts, final int port,
+            final int sslport) throws IOException {
+            super(
+                2, "front", THREADS, "back",
+                        new ServerSocket(port),
```

Comment:
@simonjenga this line (and two lines below) have to much indentation, it can be on the same level as `2`